protocols/direct-fund/direct-fund_test.go

```diff
@@ -76,6 +77,44 @@ func TestNew(t *testing.T) {
 	}
 }
 
+func TestSignPreFundSetup(t *testing.T) {
+	// Construct various variables for use in TestUpdate
+	var s, _ = New(false, testState, testState.Participants[0])
+
+	// Approve the objective, so that the rest of the test cases can run.
+	o := s.Approve()
+
+	_, se, _, err := o.Crank(&alice.privateKey)
+	if err != nil {
+		t.Error(err)
+	}
+
+	signMessage := se.MessagesToSend[0]
+
+	if signMessage.To != bob.address {
+		t.Error("incorrect recipient")
+	}
+	if signMessage.ObjectiveId != o.Id() {
+		t.Error("incorrect objective id")
+	}
+	if stateLength := len(signMessage.SignedStates); stateLength != 1 {
+		t.Error("incorrect number of Signed States")
+	}
+
+	messageSig, err := signMessage.SignedStates[0].GetParticipantSignature(s.C.MyIndex)
+	if err != nil {
+		t.Error(err)
+	}
+	stateSig, err := s.C.SignedStateForTurnNum[0].GetParticipantSignature(s.C.MyIndex)
+	if err != nil {
+		t.Error(err)
+	}
+	if sigMatch := reflect.DeepEqual(messageSig, stateSig); !sigMatch {
```

Comment:
Here's an alternative pattern using [`go-cmp`](https://github.com/google/go-cmp), which we are already using:

https://github.com/statechannels/go-nitro/blob/3c4925d201fe18bf5fe768b107565e8b3f4a7c47/channel/channel_test.go#L118-L120

Reply:
Ah that's much cleaner! I've switched over to that pattern in e5ae411